core/trino-main/src/test/java/io/trino/sql/planner/TestInsert.java

```diff
@@ -188,13 +189,42 @@ public void testCreateTableAsSelectUnpartitioned()
                                         values("column1", "column2")))));
     }
 
+    @Test
+    public void testCreateTableAsSelectWithPreferredPartitioningThreshold()
```

Comment:
this also deserves rule testing with stats, see `io.trino.sql.planner.iterative.rule.TestReorderJoins` for example